testcases/kernel/mem/hugetlb/hugeshmget/hugeshmget03.c

```diff
@@ -118,14 +118,14 @@ void setup(void)
 	tst_tmpdir();
 
 	orig_hugepages = get_sys_tune("nr_hugepages");
+	SAFE_FILE_SCANF(NULL, PATH_SHMMNI, "%ld", &orig_shmmni);
+
 	set_sys_tune("nr_hugepages", hugepages, 1);
-	hpage_size = read_meminfo("Hugepagesize:") * 1024;
+	SAFE_FILE_PRINTF(NULL, PATH_SHMMNI, "%ld", hugepages / 2);
 
+	hpage_size = read_meminfo("Hugepagesize:") * 1024;
 	shm_size = hpage_size;
 
-	SAFE_FILE_SCANF(NULL, PATH_SHMMNI, "%ld", &orig_shmmni);
-	SAFE_FILE_PRINTF(NULL, PATH_SHMMNI, "%ld", hugepages / 2);
-
 	/*
```

Comment:
I understand that you want to move the ```SAFE_FILE_SCANF``` before the ```set_sys_tune``` here but why are you moving the ```SAFE_FILE_PRINTF``` as well?